src/sentry/static/sentry/app/components/modals/sentryAppPublishRequestModal.tsx

```diff
@@ -0,0 +1,155 @@
+import {Body, Header} from 'react-bootstrap/lib/Modal';
+import styled from 'react-emotion';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {addErrorMessage, addSuccessMessage} from 'app/actionCreators/indicator';
+import _ from 'lodash';
+import {Location} from 'history';
+
+import {SentryApp} from 'app/types';
+import {t} from 'app/locale';
+import Form from 'app/views/settings/components/forms/form';
+import FormModel from 'app/views/settings/components/forms/model';
+import JsonForm from 'app/views/settings/components/forms/jsonForm';
+import space from 'app/styles/space';
+
+class PublishRequestFormModel extends FormModel {
+  getTransformedData() {
+    const data = this.getData();
+    //map object to list of questions
+    const questionnaire = Array.from(this.fieldDescriptor.values()).map(field => {
+      //we read the meta for the question that has a react node for the label
+      return {
+        question: field.meta || field.label,
+        answer: data[field.name],
+      };
+    });
+    return {questionnaire};
+  }
+}
+
+type Props = {
+  app: SentryApp;
+  location: Location;
+  closeModal: () => void;
+};
+
+export default class SentryAppPublishRequestModal extends React.Component<Props> {
+  static propTypes = {
+    app: PropTypes.object.isRequired,
+  };
+
+  form = new PublishRequestFormModel();
+
+  get formFields() {
+    const {app} = this.props;
+    //replace the : with a . so we can reserve the colon for the question
+    const scopes = app.scopes.map(scope => scope.replace(/:/, '-'));
+    const scopeQuestionBaseText =
+      'Please justify why you are requesting each of the following scopes: ';
+    const scopeQuestionPlainText = `${scopeQuestionBaseText}${scopes.join(', ')}.`;
+
+    const scopeLabel = (
+      <span>
+        {scopeQuestionBaseText}
+        {scopes.map((scope, i) => (
+          <React.Fragment key={i}>
```

Comment:
Generally you should avoid using array index as a key, you can use `scope` instead